service/forward_service.cc

```diff
@@ -119,6 +119,49 @@ class partition_ranges_owned_by_this_shard {
     }
 };
 
+// `retrying_dispatcher` is a class that dispatches forward_requests to other
+// nodes. In case of a failure, one retry is available. Request being retried
+// is executed on the super-coordinator.
+class retrying_dispatcher {
+    forward_service& _forwarder;
+    tracing::trace_state_ptr _tr_state;
+    std::optional<tracing::trace_info> _tr_info;
+    bool _retry_available = true;
```

Comment:
Why just one retry?


With RF=5 (or CL=ONE) we can tolerate two failures.

I think it's better to allow any number of retries. If we can't achieve quorum, the sub-queries will fail quickly.

Reply:
> Why just one retry?

I was afraid that allowing any number of retries would increase the query latency, so I targeted a specific scenario with only one concurrent failure. 

After some reflection I think that failing the whole query because of two node failures is much worse than introducing some latency. I will allow any number of retries in v2.

Reply:
Aren't those retries happening in parallel? So the latency is not (cost of failure) * (number of failures), it's just (cost of failure).

It will load the super-coordinator more though.

Reply:
They are happening in parallel. My reasoning was that as the load on the super-coordinator increases so does the latency.